views/detail_views.py

```diff
@@ -1,52 +1,87 @@
-import stemp_abw.models as models
+import pandas as pd
 from django.views.generic import DetailView
-from stemp_abw.app_settings import LABELS, LAYER_REGION_METADATA, LAYER_AREAS_METADATA
+
+import stemp_abw.models as models
+from meta.models import Source
 from stemp_abw import visualizations
+from stemp_abw.app_settings import LABELS, LAYER_REGION_METADATA, \
+    LAYER_AREAS_METADATA
 from wam.settings import SESSION_DATA
-from meta.models import Source
-import pandas as pd
 
 
 class MasterDetailView(DetailView):
     mode = None
     template_name = 'stemp_abw/popups/base_layer_popup.html'
     context_object_name = 'layer'
 
-    def get_context_data(self, **kwargs):
-        context = super(MasterDetailView, self).get_context_data(**kwargs)
-
-        context['title'] = LABELS['layers'][self.model.name]['title']
-        context['text'] = LABELS['layers'][self.model.name]['text']
-
-        for layer_group in LAYER_REGION_METADATA.values():
+    def get_source_data(self, metadata, app_name):
+        """
+        This method takes a metadata ConfigObj and returns a list
+        with 0 OR n-amount of Source objects, if primary keys (PK)s
+        of sources records in database are provided in ConfigObj object.
+        Values in the metadata config file should correspond to PKs
+        as list of values (1,2,3,...n). if the sole value 0 is provided
+        in the metadata config file then the returned list is empty.
+
+        Parameters
+        ----------
+        metadata : :obj:`ConfigObj`
+        app_name : :obj:`str`
+
+        Returns
+        -------
+        :obj:`list` of :obj:`Source`
```

Comment:
References to non-builtin classes should be done using
```
:class:`<path>`
```
in this case
```
:class:`wam.meta.models.Source`
```

Reply:
Thanks for the info, as spoken I will use this solution but change `:class:` to `:obj:`.